src/availability.rs

```diff
@@ -49,6 +49,19 @@ pub struct Options {
     #[arg(long = "availability-api-path", env = "HOTSHOT_AVAILABILITY_API_PATH")]
     pub api_path: Option<PathBuf>,
 
+    /// Timeout for failing requests due to missing data.
+    ///
+    /// If data needed to respond to a request is missing, it can (in some cases) be fetched from an
+    /// external provider. This parameter controls how long the request handler will wait for
+    /// missing data to be fetched before giving up and failing the request.
+    #[arg(
```

Comment:
Not for this PR, but I think that timeout logic should eventually be handled at the tide-disco level.

Reply:
Hmm. It is (actually the underlying http client (surf/tide) has a timeout) but this is different from a general request timeout. This is "we don't have the data and couldn't get it within a short time frame, so we're explicitly returning an error", and ultimately the difference is we return 404 instead of 408